torch/distributed/fsdp/_shard_utils.py

```diff
@@ -142,10 +142,10 @@ def _create_chunk_dtensor(
     corresponding chunk as the local tensor to create a DTensor.
     """
     shard_placement = DShard(0)
-    tensor_padded_list, _ = shard_placement._split_tensor(
+    tensor_list, _ = shard_placement._split_tensor(
         tensor,
         device_mesh.size(dim=0),
         with_padding=False,
         contiguous=True,
     )
-    return DTensor.from_local(tensor_padded_list[rank], device_mesh, [shard_placement])
+    return DTensor.from_local(tensor_list[rank].clone(), device_mesh, [shard_placement])
```

Comment:
Let's add a comment why clone is required. Also, not sure if we need to call `detach()`. State dict hooks may be called inside `no_grad` context. But it is probably safer to call `detach()`.